gui_handlers.go

```diff
@@ -117,7 +119,21 @@ func (ui *Ui) ShowPage(name string) {
 }
 
 func (ui *Ui) Quit() {
-	// TODO savePlayQueue/getPlayQueue
+	if len(ui.queuePage.queueData.playerQueue) > 0 {
+		ids := make([]string, len(ui.queuePage.queueData.playerQueue))
+		for i, it := range ui.queuePage.queueData.playerQueue {
+			ids[i] = it.Id
+		}
+		// stmps always only ever plays the first song in the queue
+		pos := ui.player.GetTimePos()
+		if err := ui.connection.SavePlayQueue(ids, ids[0], int(pos)); err != nil {
+			log.Printf("error stashing play queue: %s", err)
+		}
+	} else {
+		// The only way to purge a saved play queue is to force an error by providing
+		// bad data. Therefore, we ignore errors.
+		_ = ui.connection.SavePlayQueue([]string{"XXX"}, "XXX", 0)
```

Comment:
This was an attempt to circumvent the utter lack of the API to allow for clearing the queue. I tried to trick it by giving it a song ID that doesn't exist. It may work on Navidrome, but gonic verifies the IDs and simply ignores it. The error response is irrelevant, because this could _should_ be clearing the saved queue, which is impossible with the current API.